manifests/init.pp

```diff
@@ -367,7 +367,7 @@
   Array $ssl_ciphers                                                                               = [],
   Boolean $stomp_ensure                                                                            = false,
   Boolean $ldap_auth                                                                               = false,
-  String $ldap_server                                                                              = 'ldap',
+  Variant[String,Array[String]] $ldap_server                                                       = 'ldap',
```

Comment:
You can even enforce the minimal length of the string to prohibit empty strings:
```suggestion
  Variant[String[1],Array[String[1]]] $ldap_server                                                 = 'ldap',
```